### ISSUE TITLE
java.lang.AbstractMethodError: io.github.spark_redshift_community.spark.redshift.RedshiftFileFormat.shouldPrefetchData(Lorg/apache/spark/sql/SparkSession;Lorg/apache/spark/sql/types/StructType;Lorg/apache/spark/sql/types/StructType;)Z

### ISSUE THREAD

**Issue by pmalafosse** [spark-redshift-community/spark-redshift#61]
Hello,

I'm using EMR 5.28.0 with Spark 2.4.4 and Scala 2.11.12 and I get this error when I try to read a Redshift table. Does anyone know how to fix it please?

```java.lang.AbstractMethodError: io.github.spark_redshift_community.spark.redshift.RedshiftFileFormat.shouldPrefetchData(Lorg/apache/spark/sql/SparkSession;Lorg/apache/spark/sql/types/StructType;Lorg/apache/spark/sql/types/StructType;)Z```

I do the following:
`spark-shell --jars RedshiftJDBC42-no-awssdk-1.2.37.1061.jar --packages io.github.spark-redshift-community:spark-redshift_2.11:4.0.0`

sql("""CREATE temporary view users_rs (`id` INT, `name` STRING)
USING io.github.spark_redshift_community.spark.redshift
OPTIONS (
  `tempdir` 's3a://mybucket/users/',
  `url` 'jdbc:redshift://host:5439/database?user=<user>&password=<password>',
  `forward_spark_s3_credentials` 'true',
  `serialization.format` '1',
  `dbtable` 'schema.users'
)
""")

sql("SELECT * FROM users_rs where id = 10000").show()

FYI I had found about this fork there: https://github.com/databricks/spark-redshift/issues/417 but unfortunately I get the same error.

**Comment by pmalafosse** [comment]
I also tried launching Spark like that but get the same error:
`spark-shell --jars RedshiftJDBC42-n-community:spark-redshift_2.11:4.0.0,org.apache.spark:spark-avro_2.11:2.4.4`

**Comment by k0l0nell** [comment]
Just ran into that as well but couldn't quite trace it all the way through to the root cause. I "fixed" this by adding a `shouldPrefetchData` method to `RedshiftFileFormat.scala` 

```scala
def shouldPrefetchData(sparkSession: SparkSession,s1: StructType,s2: StructType): Boolean = false
```

Can't comment on possible side effects of this though. Need to find some time to see where  shouldPrefetchData comes from as I can't find it in spark sql sources

**Comment by lucagiovagnoli** [comment]
All this is hard to reproduce for me because we don't have testing on EMR and we don't use this package in EMR at Yelp. Where does shouldPrefetchData  come from?

@k0l0nell if you'd like to open a PR on here, I'll be happy to review.

**Comment by k0l0nell** [comment]
> All this is hard to reproduce for me because we don't have testing on EMR and we don't use this package in EMR at Yelp. Where does shouldPrefetchData come from?

No idea where it comes from or what makes the call. I will follow this through with AWS as I can't find this in vanilla spark.

> @k0l0nell if you'd like to open a PR on here, I'll be happy to review.

Won't just yet until I understand what the root cause is :)

**Comment by pmalafosse** [comment]
FYI this bug was fixed in EMR from 5.28.1 version.
